File: 프로그래머스/0/181894. 2의 영역/2의 영역.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> index = new ArrayList<>();
        List<Integer> answer = new ArrayList<>();
        
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 2) {
                index.add(i);
            }
        }
        
        if (index.size() == 0) {
            return new int[] {-1};
        }
        
        if (index.size() == 1) {
            return new int[] {2};
        }
        
        for (int i = index.get(0); i < index.get(index.size() - 1) + 1; i++) {
            answer.add(arr[i]);
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/0/181894. 2의 영역/2의 영역.py
def solution(arr):
    startIdx = -1
    endIdx = -1
    for i in range(len(arr)):
        if arr[i] == 2:
            startIdx = i
            break
    
    for i in range(len(arr) - 1, -1, -1):
        if arr[i] == 2:
            endIdx = i
            break
            
    if startIdx == -1:
        return [-1]
            
    if endIdx == -1:
        return arr[startIdx:]
    
    return arr[startIdx:endIdx+1]